dynamics/src/include/DynamicsKernel.hpp

```diff
@@ -181,6 +171,52 @@ template <int DGadvection, int DGstress> class DynamicsKernel {
         Nextsim::LimitMin(hice, 0.0);
     }
 
+    /*!
+     * Prepares the transport objects to perform the advection step.
+     */
+    virtual void prepareAdvection() = 0;
+
+    /*!
+     * A wrapper to advect a ModelArray field one timestep.
+     *
+     *
+     * @param timestep The advection timestep in seconds.
+     * @param field A reference to the field to be advected.
+     * @param lowerLimit The minimum value the field is allowed to take. Defaults to -∞.
+     * @param upperLimit The maximum value the field is allowed to take. Defaults to +∞.
+     *
+     * @return A reference to the advected array.
+     */
+    ModelArray& advectField(double timestep, ModelArray& field,
+        double lowerLimit = -std::numeric_limits<double>::infinity(),
+        double upperLimit = std::numeric_limits<double>::infinity())
+    {
+        DGVectorHolder<DGadvection> holder(field);
+        advectDGVField(timestep, holder, lowerLimit, upperLimit);
+        return field;
+    }
+
+    /*!
+     * Advects a field one timestep.
+     *
+     * @param timestep The advection timestep in seconds.
+     * @param field A reference to the field to be advected.
+     * @param lowerLimit The minimum value the field is allowed to take. Defaults to -∞.
+     * @param upperLimit The maximum value the field is allowed to take. Defaults to +∞.
+     *
+     * @return A reference to the advected array.
+     */
+    virtual DGVector<DGadvection>& advectDGVField(double timestep, DGVector<DGadvection>& field,
+        double lowerLimit = -std::numeric_limits<double>::infinity(),
+        double upperLimit = std::numeric_limits<double>::infinity())
+        = 0;
+
+    virtual void advectDynamicsFields(double timestep)
+    {
+        advectDGVField(timestep, hice, 0.0);
```

Comment:
```suggestion
        advectDGVField(timestep, hice, 0.0);
        advectDGVField(timestep, hsnow, 0.0);
```

is the way to go. We also need to use the total mass of the ice and snow in the momentum equation, but that's for another PR

Reply:
Done.